### DISCUSSION TITLE
Rebasing instead of merging

### DISCUSSION THREAD

**Discussion by maximeborges** [stm32-rs/stm32f7xx-hal#188] [General]
I propose to switch to rebasing by default, so we get a nice and clean commit history which will be easier to navigate through.
Most of the PRs we get are just a few commits, most of time actually just one commit.
Having merge commits double that number for nothing, and makes commit graphs  actually even worse to look at.
![image](https://user-images.githubusercontent.com/159235/172069285-b1517121-2c7d-49ad-bfcd-e145304086a0.png)

**Comment by eldruin** [comment]
Sounds good to me, let's do that.
